cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -710,9 +710,12 @@ private synchronized void finished() {
                 @Override
                 public void run() {
                     try {
-                        int c;
-                        while ((c = System.in.read()) != -1) { // TODO use InputStream.available
-                           stdin.write(c);
+                        int available = System.in.available();
+                        if (available != 0) {
+                            byte[] bytes = new byte[available];
```

Comment:
The javadoc of the available method says: _. It is never correct to use the return value of this method to allocate a buffer intended to hold all data in this stream._ I think it could fail depending on the implementation. Anyway it's not recommended according the javadoc. Do you think it is worth to consider any other alternative?

Reply:
Good catch!. I knew the method but I didn't know it's discouraged. I'll still work on it.